sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -506,20 +533,23 @@ async Task GetScopeProperties (MessageId msg_id, int scope_id, CancellationToken
 
 				var values = res.Value? ["result"]? ["value"]?.Values<JObject> ().ToArray ();
 
-				var var_list = new List<JObject> ();
+				if(values == null)
+					SendResponse (msg_id, Result.OkFromObject (new {result = Array.Empty<object> ()}), token);
+
+				var var_list = new List<object> ();
 				int i = 0;
 				// Trying to inspect the stack frame for DotNetDispatcher::InvokeSynchronously
 				// results in a "Memory access out of bounds", causing 'values' to be null,
 				// so skip returning variable values in that case.
-				while (values != null && i < vars.Length && i < values.Length) {
+				while (i < vars.Length && i < values.Length) {
```

Comment:
replace with a `for (int i = 0; i < vars.Length && i < values.Length; i ++) {` ?

Reply:
Since i is used without being reset in the next loop as well I'd prefer to keep the scope clear

Reply:
ah, missed that! 👍